src/platform_impl/linux/mod.rs

```diff
@@ -519,6 +519,16 @@ impl Window {
         x11_or_wayland!(match self; Window(w) => w.set_ime_allowed(allowed))
     }
 
+    #[inline]
+    pub fn set_ime_purpose(&self, _purpose: ImePurpose) {
+        match self {
+            #[cfg(x11_platform)]
+            Window::X(_) => (),
+            #[cfg(wayland_platform)]
+            Window::Wayland(ref w) => w.set_ime_purpose(_purpose),
```

Comment:
You should add a blank impl for X11 and use the macro, that way it should be consistent with the rest of the calls.